code/modules/admin/DB ban/functions.dm

```diff
@@ -95,10 +59,9 @@
 		a_computerid = src.owner:computer_id
 		a_ip = src.owner:address
 
-	if(blockselfban)
-		if(a_ckey == ckey)
-			usr << "<span class='danger'>You cannot apply this ban type on yourself.</span>"
-			return
+	if(!job && (a_ckey == ckey))
+		usr << "<span class='danger'>You cannot server ban yourself.</span>"

```

Comment:
If removed, a warning prompt would be good; we all know someone will eventually accidentally ban themselves otherwise.